### PR TITLE
test(models): Added unit cases for Permissions, LicenseCandidate, Gro…

### PR DESCRIPTION
**Pull request by valens200**
## Description
 This pull request introduces more unit test coverage for the models like Permissions, LicenseCandidate, Group, Decider,  FileInfo and Agent as reported in issue [#2755](https://github.com/fossology/fossology/issues/2773)

### Changes
**Enhanced existing test cases:**  Worked on the  current test suite to improve the coverage.
**Added new test cases:** Introduced additional tests to  validate all methods and member variables.

### How to Test ?

- Navigate to the ui_tests/api/models directory.
- Execute each test case individually.
- Verify that all tests pass without any errors or failures.

CC @shaheemazmalmmd @soham4abc @dushimsam @GMishx

### CODE REVIEW

File: src/www/ui/api/Helper/DbHelper.php [hunk 1]
```diff
@@ -310,7 +310,7 @@ public function getJobs($id = null, $limit = 0, $page = 1, $uploadId = null)
 
     $params = [];
     $statement = __METHOD__ . ".getJobs";
-    $countStatement = __METHOD__ . ".getJobCount";
+    $count  Statement = __METHOD__ . ".getJobCount";
```
Comment (GMishx) [inline src/www/ui/api/Helper/DbHelper.php:313]: Uhho. Please fix the variable name.

File: src/www/ui/api/Helper/DbHelper.php [hunk 2]
```diff
@@ -491,7 +491,7 @@ public function insertNewTokenKey($userId, $expire, $scope, $name, $key)
    * Adds new oauth client to the user.
    *
    * @param string  $name     Name of the new client
-   * @param integer $userId   User PK
+   * @param integer￼ $userId   User PK
```
Comment (GMishx) [inline src/www/ui/api/Helper/DbHelper.php:494]: Irrelevant change. Please remove.

File: src/www/ui/page/UploadFilePage.php [hunk 3]
```diff
@@ -15,6 +15,7 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
+
```
Comment (GMishx) [inline src/www/ui/page/UploadFilePage.php:18]: Irrelevant change. Please remove.

File: src/www/ui_tests/api/Models/AgentTest.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ SPDX-FileCopyrightText: © 2020 Siemens AG
+ Author: Valens Niyonsenga <valensniyonsenga2003@gmail.com>
```
Comment (GMishx) [inline src/www/ui_tests/api/Models/AgentTest.php:5]: Please fix the copyright year and remove Siemens AG name. Use yours instead.

File: src/www/ui_tests/api/Models/FileInfoTest.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ SPDX-FileCopyrightText: © 2020 Siemens AG
+ Author: Valens Niyonsenga <valensniyonsenga2003@gmail.com>
```
Comment (GMishx) [inline src/www/ui_tests/api/Models/FileInfoTest.php:5]: Please fix the copyright year and remove Siemens AG name. Use yours instead.

File: src/www/ui_tests/api/Models/GroupTest.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
+<?php
+/*
+ SPDX-FileCopyrightText: © 2020 Siemens AG
+ Author: Valens Niyonsenga <valensniyonsenga2003@gmail.com>
```
Comment (GMishx) [inline src/www/ui_tests/api/Models/GroupTest.php:4]: Please fix the copyright year and remove Siemens AG name. Use yours instead.

File: src/www/ui_tests/api/Models/LicenseCandidateTest.php [hunk 7]
```diff
@@ -0,0 +1,186 @@
+<?php
+
+/*
+ SPDX-FileCopyrightText: © 2020 Siemens AG
+ Author: Valens Niyonsenga <valensniyonsenga2003@gmail.com>
```
Comment (GMishx) [inline src/www/ui_tests/api/Models/LicenseCandidateTest.php:5]: Please fix the copyright year and remove Siemens AG name. Use yours instead.

File: src/www/ui_tests/api/Models/PermissionsTest.php [hunk 8]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ SPDX-FileCopyrightText: © 2020 Siemens AG
+ Author: Valens Niyonsenga <valensniyonsenga2003@gmail.com>
```
Comment (GMishx) [inline src/www/ui_tests/api/Models/PermissionsTest.php:5]: Please fix the copyright year and remove Siemens AG name. Use yours instead.

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/www/ui/api/Helper/DbHelper.php (hunk 1 above)
Comment (valens200) [reply src/www/ui/api/Helper/DbHelper.php]: Thank you @GMishx , the naming is now fixed.

File: src/www/ui/api/Helper/DbHelper.php (hunk 2 above)
Comment (valens200) [reply src/www/ui/api/Helper/DbHelper.php]: Thank you @GMishx , irrelevant change is now removed.

File: src/www/ui/page/UploadFilePage.php (hunk 3 above)
Comment (valens200) [reply src/www/ui/page/UploadFilePage.php]: Removed ,  thank you!

File: src/www/ui_tests/api/Models/GroupTest.php (hunk 6 above)
Comment (valens200) [reply src/www/ui_tests/api/Models/GroupTest.php]: Done, thank you so much!.

File: src/www/ui_tests/api/Models/PermissionsTest.php (hunk 8 above)
Comment (valens200) [reply src/www/ui_tests/api/Models/PermissionsTest.php]: Done, thank you!

File: src/www/ui_tests/api/Models/AgentTest.php (hunk 4 above)
Comment (valens200) [reply src/www/ui_tests/api/Models/AgentTest.php]: Done, thank you!

**Review by GMishx — APPROVED** [review]
Changes looks good

File: src/www/ui/api/Helper/DbHelper.php [hunk 9]
```diff
@@ -491,7 +491,7 @@ public function insertNewTokenKey($userId, $expire, $scope, $name, $key)
    * Adds new oauth client to the user.
    *
    * @param string  $name     Name of the new client
-   * @param integer $userId   User PK
+   * @param integer￼$userId  User PK
```
Comment (dvjsharma) [inline src/www/ui/api/Helper/DbHelper.php:494]: Irrelevant change, please revert.

File: src/www/ui/api/Models/Agent.php [hunk 10]
```diff
@@ -122,16 +122,28 @@ public function getJSON($version=ApiVersion::V1)
...
+      return [
+        'successfulAgents' => $this->getSuccessfulAgents(),
+        'uploadId' => $this->getUploadId(),
+        'agentName' => $this->getAgentName(),
+        'currentAgentId' => $this->getCurrentAgentId(),
+        'currentAgentRev' => $this->getCurrentAgentRev(),
+        'isAgentRunning' => $this->getIsAgentRunning()
+      ];
+    }
+
```
Comment (dvjsharma) [inline src/www/ui/api/Models/Agent.php:146]: The V1 `return` keys do not exist, and neither are required (`successfulagents` and so on). Only the version needs to be passed in `getArray()`, and the model return syntax remains unchanged. Please revert.

File: src/www/ui/api/Models/Agent.php [hunk 11]
```diff
@@ -181,4 +193,4 @@ public function setIsAgentRunning($isAgentRunning)
   {
     $this->isAgentRunning = $isAgentRunning;
   }
-}
\ No newline at end of file
+}
```
Comment (dvjsharma) [inline src/www/ui/api/Models/Agent.php:196]: Irrelevant change, please revert.

File: src/www/ui_tests/api/Models/AgentTest.php [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+        'uploadId' => 4,
+        'agentName' => "decisionimporter",
+        'currentAgentId' => 24,
+        'currentAgentRev' => "68097a",
+        'isAgentRunning' => null
+      ];
+    }
+    return [
+      'agentInfo' =>  $agentInfo,
+      'obj' => new Agent([], 4, "decisionimporter", 24, "68097a", null)
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/AgentTest.php:47]: Ditto, tests will need to be adjusted due to a change in the agent model as described above. (No valid key in V1 `return`).

File: src/www/ui_tests/api/Models/DeciderTest.php [hunk 13]
```diff
@@ -11,116 +11,131 @@
  */
 
 namespace Fossology\UI\Api\Test\Models;
-
-use Fossology\UI\Api\Models\Decider;
 use Fossology\UI\Api\Models\ApiVersion;
+use Fossology\UI\Api\Models\Decider;
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/DeciderTest.php:15]: Irrelevant change, please revert.

File: src/www/ui_tests/api/Models/DeciderTest.php [hunk 14]
```diff
@@ -11,116 +11,131 @@
...
-    } else {
-      $deciderArray = [
-        "nomosMonk" => true,
+    }else{
+      $deciderInfo = [
+        "nomosMonk"  => true,
         "bulkReused" => false,
-        "ojoDecider" => (1==1)
+        "newScanner" => false,
+        "ojoDecider" => true
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/DeciderTest.php:74]: No changes are needed as this test already ran successfully for both V1 and V2. If we already have both version tests explicitly defined, I don't think we should remove them until V1 is completely depricated.

Comment (dvjsharma) [inline src/www/ui_tests/api/Models/DeciderTest.php:74]: No changes are needed as this test already ran successfully for both V1 and V2. If we already have both version tests explicitly defined, I don't think we should remove them until V1 is completely deprecated.

File: src/www/ui_tests/api/Models/DeciderTest.php [hunk 15]
```diff
@@ -11,116 +11,131 @@
...
-        "ojoDecider" => true
-      ];
-    }
-
-    $actualObject = new Decider();
-    $actualObject->setNomosMonk(true);
-    $actualObject->setOjoDecider(true);
+    $this->decider->setNewScanner(true);
+    $this->assertTrue($this->decider->getNewScanner());
+  }
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/DeciderTest.php:129]: Ditto

File: src/www/ui_tests/api/Models/FileInfoTest.php [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+  /**
+   * Test FileInfo:getArray() with ApiVersion::V1
+   * Tests the getArray method of the FileInfo class with API version 1.
+   *  - # Check if the returned array matches the expected data for version 1
+   */
+  public function testDataFormatVersion1()
+  {
+    $expectedArray = $this->getDummyData()['data'];
+    $obj = $this->getDummyData()['obj'];
+    $actual = $obj->getArray(ApiVersion::V2);
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/FileInfoTest.php:75]: Should be `ApiVersion::V1` since test is for V1.

File: src/www/ui_tests/api/Models/GroupTest.php [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+class GroupTest extends \PHPUnit\Framework\TestCase
+{
+  /**
+   * Provides test data and an instance of the Group class.
+   *
+   * @return array An associative array containing test data and a Group object.
+   */
+  public function getGroupInfo()
+  {
+    $v1 = [
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/GroupTest.php:24]: There is no versioning difference in this model. Maybe the name  can be changed to a more general name than `$v1`.

File: src/www/ui_tests/api/Models/LicenseCandidateTest.php [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+      $candidateInfo = [
+        'id' => $id,
+        'shortname' => $shortname,
+        'spdxid' => $spdxid,
+        'fullname' => $fullname,
+        'text' => $text,
+        'group_name' => $group_name,
+        'group_id' => $group_id,
+      ];
+    }
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/LicenseCandidateTest.php:54]: keys are wrong, please check with the `LicenseCandidate` model.

File: src/www/ui_tests/api/Models/UploadTest.php [hunk 19]
```diff
@@ -77,6 +77,7 @@ private function testDataFormat($version)
 
     $actualUpload = new Upload(2, 'root', 3, '', 'my.tar.gz', '01-01-2020', 3,
       $hash);
+    var_dump($actualUpload->getArray());
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/UploadTest.php:80]: Why is this required?

**Review by dvjsharma — CHANGES_REQUESTED** [review]
Hey @valens200. Left some comments, PTAL.

File: src/www/ui_tests/api/Models/LicenseCandidateTest.php (hunk 18 above)
Comment (valens200) [reply src/www/ui_tests/api/Models/LicenseCandidateTest.php:54]: For V1, they sould be separated by underscores.

File: src/www/ui_tests/api/Models/FileInfoTest.php (hunk 16 above)
Comment (valens200) [reply src/www/ui_tests/api/Models/FileInfoTest.php:75]: Sure , resolved.

**Comment by valens200** [comment]
> Hey @valens200. Left some comments, PTAL.

Sure @dvjsharma , let me know if changes looks good.

File: src/www/ui/api/Models/Agent.php (hunk 10 above)
Comment (valens200) [reply src/www/ui/api/Models/Agent.php]: resolved

File: src/www/ui_tests/api/Models/UploadTest.php (hunk 19 above)
Comment (valens200) [reply src/www/ui_tests/api/Models/UploadTest.php]: this one removed.

File: src/www/ui/api/Helper/DbHelper.php [hunk 20]
```diff
@@ -491,7 +491,7 @@ public function insertNewTokenKey($userId, $expire, $scope, $name, $key)
    * Adds new oauth client to the user.
    *
    * @param string  $name     Name of the new client
-   * @param integer $userId   User PK
+   * @param integer￼$userId   User PK
    * @param string  $clientId New client ID
```
Comment (dvjsharma) [inline src/www/ui/api/Helper/DbHelper.php:495]: Irrelevant change, please revert.

File: src/www/ui/api/Models/Decider.php [hunk 21]
```diff
@@ -151,22 +151,23 @@ public function setOjoDecider($ojoDecider)
...
         "ojo_decider" => $this->ojoDecider
       ];
+    }else{
+      return [
+        "nomosMonk"  => $this->nomosMonk,
+        "bulkReused" => $this->bulkReused,
+        "newScanner" => $this->newScanner,
+        "ojoDecider" => $this->ojoDecider
+      ];
     }
```
Comment (dvjsharma) [inline src/www/ui/api/Models/Decider.php:170]: Consider using V1 as the default pattern. I followed the same during upgrades. (`if` should check for `V2`, and `else` will handle all other cases).

File: src/www/ui_tests/api/Controllers/GroupControllerTest.php [hunk 22]
```diff
@@ -108,7 +108,7 @@ protected function setUp() : void
    */
   protected function tearDown() : void
   {
-    $this->addToAssertionCount(
+    $this->addToAssertionCount(
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Controllers/GroupControllerTest.php:111]: Irrelevant change, please revert.

File: src/www/ui_tests/api/Models/DeciderTest.php [hunk 23]
```diff
@@ -11,65 +11,72 @@
  */
 
 namespace Fossology\UI\Api\Test\Models;
-
-use Fossology\UI\Api\Models\Decider;
 use Fossology\UI\Api\Models\ApiVersion;
+use Fossology\UI\Api\Models\Decider;
```
Comment (dvjsharma) [inline src/www/ui_tests/api/Models/DeciderTest.php:15]: Irrelevant change, please revert.

**Review by dvjsharma — CHANGES_REQUESTED** [review]
Hi @valens200. Minor changes, rest looks good.

Also, I see you have squashed other commits as well (the `Dockerfile` change for example), please fix them.
